lib/coordinatorsharding.go

```diff
@@ -386,8 +387,9 @@ func (crd *Coordinator) PreprocessSharding(requests []*netstring.Netstring) (boo
 						}
 						evt := cal.NewCalEvent(EvtTypeSharding, EvtNameNoShardKey, cal.TransOK, "")
 						evt.AddDataInt("sql", int64(uint32(crd.sqlhash)))
+						evt.AddDataStr("shard_key", GetConfig().ShardKeyName)
 						evt.Completed()
-						ns := netstring.NewNetstringFrom(common.RcError, []byte(ErrNoShardKey.Error()))
+						ns := netstring.NewNetstringFrom(common.RcError, []byte(fmt.Sprintf(ErrNoShardKey.Error(), GetConfig().ShardKeyName)))
```

Comment:
I'd suggest we append the shard key to the RcError message than having a placeholder in ErrNoShardKey. I'd prefer the error to be constant.

Reply:
Done